src/sort/mod.rs

```diff
@@ -102,6 +104,16 @@ pub trait Sort: Any + Send + Sync + Debug {
     }
 }
 
+pub trait Presort {
+    fn name() -> Symbol;
```

Comment:
Optional: both Presort and Sort have the `name()` methods. This can be confusing as they are used for different purposes.

Reply:
I agree. I was going to use `Presort::presort_name` instead of `Presort::name`, but that conflicts with `Presort::presort_names`. Maybe the problem is `TypeInfo.presort_names`? Is there a better name for that?

Reply:
Conclusion: change `presort_names` to `reserved_primitives` and add a check to make sure that primitives don't get reserved twice.